Reformat BigQuery connector requirements section (#8412)
trinodb/trino: docs/src/main/sphinx/connector/bigquery.rst

Requirements
------------

* Enable the `BigQuery Storage Read API

>   <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.

Reviewer: After this add the network access item from coordinator and workers. Not sure about any default port..

Author: Since this connector uses the Google Cloud API, it should just be hitting their Cloud project's service endpoint (which, as best I can tell, uses HTTPS without any custom port by default).

Reviewer 2: This reminds me - do we document anything regarding access to HDFS/S3 for the Hive connector? Or are we only mentioning the metastore there?

Reviewer: We also mention access to the storage .. its critically important for things to work..

Reviewer: And in this case we need to mention that access to the Google Cloud API from the coordinator and workers is necessary.

Reviewer 2: In that case please do mention that permissions are needed in BigQuery.
See https://cloud.google.com/bigquery/docs/reference/storage

> requires permissions to two distinct resources within BigQuery: the project that controls the session and the table from which the data is read.

Rest of the communication happens over HTTPS probably but the Google Cloud SDK doesn't call it out so I cannot be sure about that part.

Reviewer: We still need to at least mention it since you could run Trino in a private network without global internet access and then the Google Cloud API has to be added to an allowlist

Author: Rephrased slightly to be more clear about network access to the API endpoint (through which, I believe, we get access to the storage - no separate storage network access). Also added a requirement for setting BigQuery permissions.

How's that look?

* Set up authentication. Your authentiation options differ depending on whether
  you're using Dataproc/Google Compute Engine (GCE) or not.

  **On Dataproc/GCE** the authentication is taken from the machine's role.

  **Outside Dataproc/GCE** you have 3 options:

  * Use a service account JSON key and ``GOOGLE_APPLICATION_CREDENTIALS`` as

>     described in the Google authentication `getting started guide

Reviewer 2: Maybe?
```suggestion
    described in the Google Cloud `getting started guide
```

Author: Problem is it's specifically a getting started guide to authentication, not Cloud.

How about "..in the Google Cloud authentication `getting started guide..."?

Reviewer 2: "..in the Google Cloud authentication `getting started guide..." is much better. Thanks

Requirements
------------

* Enable the `BigQuery Storage Read API
  <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.
* Ensure that the Trino coordinator and workers have network access to your
  Google Cloud service endpoint.
* Set up authentication. Your authentiation options differ depending on whether

>   you're using Dataproc/Google Compute Engine (GCE) or not.

Reviewer 3: We generally try to stay away from contractions. suggest "you use" instead.

Requirements
------------

* Enable the `BigQuery Storage Read API
  <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.
* Ensure that the Trino coordinator and workers have network access to your
  Google Cloud API service endpoint.

> * Configure BigQuery so that the Trino coordinator and workers have permission
>   to `control the session and read data
>   <https://cloud.google.com/bigquery/docs/reference/storage#permissions>`_.

Reviewer 2: The wording is vague on GCP docs too. I don't know what exact permissions are required so can we instead say.

```suggestion
* Configure BigQuery so that the Trino coordinator and workers have `permissions
  to BigQuery
  <https://cloud.google.com/bigquery/docs/reference/storage#permissions>`_.
```

* Set up authentication. Your authentiation options differ depending on whether

>   you use using Dataproc/Google Compute Engine (GCE) or not.

Reviewer 2: ```suggestion
  you are using Dataproc/Google Compute Engine (GCE) or not.
```

>   **On Dataproc/GCE** the authentication is taken from the machine's role.

Reviewer 2: ```suggestion
  **On Dataproc/GCE** the authentication is done using the machine's role.
```
